src/ripple/app/main/Application.cpp

```diff
@@ -1584,7 +1584,7 @@ ApplicationImp::loadLedgerFromFile (
             ledger = ledger.get()["accountState"];
         }
 
-        if (!ledger.get().isArray ())
+        if (!ledger.get().isArrayOrNull ())
```

Comment:
I'm happy to change to `isArray()` (delighted actually).  I was hesitant because with the previous semantics a null value was considered to be a valid array of length 0 and I couldn't convince myself that a null value was not an acceptable result here (even with that log message).

But if you (or anyone else) can assure me that a null value should error out here, then this is exactly the kind of bug fixing I was hoping to uncover with this PR.

Reply:
Unrelated:

She's one weird lib this one huh :) ?  I noticed grabbing a reference via `obj["someKey"]` will initially be a null type that can later be changed to an array, merely by calling the `append(...)` method on it.